src/test/java/org/cactoos/func/BiFuncOfTest.java

```diff
@@ -23,57 +23,88 @@
  */
 package org.cactoos.func;
 
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.cactoos.BiFunc;
 import org.cactoos.proc.ProcOf;
-import org.cactoos.scalar.True;
-import org.hamcrest.core.IsEqual;
+import org.cactoos.scalar.Constant;
+import org.hamcrest.core.IsSame;
 import org.junit.jupiter.api.Test;
 import org.llorllale.cactoos.matchers.Assertion;
-import org.llorllale.cactoos.matchers.IsTrue;
+import org.llorllale.cactoos.matchers.MatcherOf;
 
 /**
  * Test case for {@link BiFuncOf}.
  *
  * @since 0.20
- * @checkstyle JavadocMethodCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 final class BiFuncOfTest {
 
     @Test
     void convertsFuncIntoBiFunc() throws Exception {
         new Assertion<>(
             "Must convert function into bi-function",
             new BiFuncOf<>(
-                new FuncOf<>(input -> 1)
-            ).apply(1, 2),
-            new IsEqual<>(1)
+                new FuncOf<>(input -> input)
+            ),
+            new MatcherOf<>(
+                func -> {
+                    final Object first = new Object();
+                    final Object res = func.apply(first, "discarded");
+                    return res.equals(first);
```

Comment:
@victornoel I think that in this context, it is better to use `==` because we actually get object `first`.